API.php

```diff
@@ -1224,8 +1224,13 @@ public function disablePreviewMode($idSite, $idContainer)
      * @param string $idContainer  The id of a container, for example "6OMh6taM"
      * @param string $url  The url to enable debug
      */
-    public function changeDebugUrl($idSite, $idContainer, $url)
+    public function changeDebugUrl($idSite, $url)
     {
+        $this->accessValidator->checkWriteCapability($idSite);
+        if (!filter_var($url, FILTER_VALIDATE_URL)) {
```

Comment:
I'm actually wondering if using `FILTER_VALIDATE_URL` only might cause problems here. It does not check if the protocol is "safe". It could also be `ssh://` or similar. So we could additionally check `preg_match('/^https?:\/\//')`, or do you think other protocols are needed there as well?

Reply:
@sgiehl Added `stripos($url, 'http')` check to ensure valid protocol is present.

Reply:
the problem with `stripos === false` is, that you don't check if it's in the beginning. So something like `ssh://http.webhost.com` would pass here... So guess you would need to compare with `!== 0`

Reply:
Good catch :+1:

Reply:
Updated and added few testcases too